File: Android_APP/app/src/main/java/com/smuemsw/capstone/AccessLogListViewAdapter.java
package com.smuemsw.capstone;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class AccessLogListViewAdapter extends BaseAdapter {

    Context mContext;
    LayoutInflater mLayoutInflater;
    ArrayList<AccessLogListViewData> itemList;

    public AccessLogListViewAdapter(Context context, ArrayList<AccessLogListViewData> data){

        mContext = context;
        itemList = data;
        mLayoutInflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return itemList.size();
    }

    @Override
    public Object getItem(int position) {
        return itemList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = mLayoutInflater.inflate(R.layout.activity_access_log_listview_item, null);

        TextView time = view.findViewById(R.id.accessLogListViewItemTime);
        TextView ip = view.findViewById(R.id.accessLogListViewItemIP);
        TextView location = view.findViewById(R.id.accessLogListViewItemLocation);

        time.setText(itemList.get(position).getTime());
        ip.setText(itemList.get(position).getIp());
        location.setText(itemList.get(position).getLocation());

        return view;
    }
}

File: checkIn/app/src/main/java/org/checkinserviceteam/android/checkin/adater/AccessLogListViewAdapter.kt
package org.checkinserviceteam.android.checkin.adater

import android.content.Context
import android.text.Layout
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import kotlinx.android.synthetic.main.item_access_log_listview.view.*
import org.checkinserviceteam.android.checkin.R
import org.checkinserviceteam.android.checkin.data.AccessLogListViewData
import java.util.ArrayList

class AccessLogListViewAdapter(
    var mContext: Context,
    var itemList: ArrayList<AccessLogListViewData>
) : BaseAdapter() {

    override fun getCount(): Int {
        return itemList.size
    }

    override fun getItem(position: Int): Any {
        return itemList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_access_log_listview, null)

        view.item_access_log_listview_tv_time.text = itemList[position].time
        view.item_access_log_listview_tv_ip.text = itemList[position].ip
        view.item_access_log_listview_tv_status.text = itemList[position].status

        return view
    }
}